backend/src/Controller/ProductController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Serializer\SerializerInterface;
+use Symfony\Component\Routing\Annotation\Route;
+use App\Repository\ProductRepository;
+use App\Entity\Product;
+
+class ProductController extends AbstractController
+{
+    /**
+     * @Route("/products", methods={"GET"})
+     */
+    public function index(
+        ProductRepository $repository, 
+        SerializerInterface $serializer): JsonResponse
+    {
+        $products = $repository->findAll();
+
+        return new JsonResponse(
+            $serializer->serialize($products, 'json'),
+            JsonResponse::HTTP_OK,
+            [],
+            true
+        );
+
+    }
+
+    /**
+     * @Route("/products/{id}", methods={"GET"})
+     */
+    public function findId(
```

Comment:
Maybe nitpicking but "getById" as a function Name would be even more precise.

Reply:
yes thats better, thank you :)